airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -2024,16 +2023,34 @@ def create_async_retriever(
             name=f"job polling - {name}",
         )
         job_download_components_name = f"job download - {name}"
+        download_decoder = (
+            self._create_component_from_model(model=model.download_decoder, config=config)
+            if model.download_decoder
+            else JsonDecoder(parameters={})
+        )
+        download_extractor = (
+            self._create_component_from_model(
+                model=model.download_extractor,
+                config=config,
+                decoder=download_decoder,
+                parameters=model.parameters,
+            )
+            if model.download_extractor
+            else DpathExtractor(
+                [], config=config, decoder=download_decoder, parameters=model.parameters
+            )
+        )
         download_requester = self._create_component_from_model(
             model=model.download_requester,
-            decoder=decoder,
+            decoder=download_decoder,
             config=config,
             name=job_download_components_name,
         )
         download_retriever = SimpleRetriever(
             requester=download_requester,
             record_selector=RecordSelector(
-                extractor=ResponseToFileExtractor(),
+                # extractor=ResponseToFileExtractor(),# old one
```

Comment:
Can we remove this line?

Reply:
sure, do we need to add `ResponseToFileExtractor` to the list of possible choices `Decoders` ?

Reply:
Do you mean "to the list of possible `extractors`"? I think that would make sense. I think we were reluctant at first because it was considered experimental since it wasn't in the Connector Builder but if we need it, I'm good with that